src/Sarif/Map/JsonMapSettings.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Sarif.Map
+{
+    /// <summary>
+    ///  JsonMapSettings contains settings to control how JsonMaps are constructed.
+    ///  For now, this is only the Map size target, to determine how much detail the map will include.
+    /// </summary>
+    public class JsonMapSettings
+    {
+        public const long NodeSizeEstimateBytes = 90;         // Bytes for one Node with containing property but not including children
+        public const long ArrayStartSizeEstimateBytes = 5;    // Bytes for one ArrayStart location, if under 10KB.
+        public const long Megabyte = 1024 * 1024;
+
+        public double CurrentSizeRatio { get; set; }
+        public int MinimumSizeForNode { get; set; }
+
+        public double MapDesiredSizeRatio { get; set; }
+        public double MapMaximumSizeBytes { get; set; }
+
+        /// <summary>
+        ///  Default Settings: Map is 1% of size of file mapped; up to a limit of 10 MB.
+        /// </summary>
+        public static JsonMapSettings DefaultSettings => new JsonMapSettings(0.01, 10 * Megabyte);
```

Comment:
> J [](start = 21, length = 2)

`readonly`

Reply:
Readonly isn't allowed here by the compiler and since I'm using =>, this is a property with no setter, so it's the functional equivalent of public static JsonMapSettings DefaultSettings { get; } with initialization in each constructor.

Reply:
Oh right, this is a get-only auto-property.

---
In reply to: [325825989](https://github.com/microsoft/sarif-sdk/pull/1672#discussion_r325825989) [](ancestors = 325825989)